packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -173,43 +206,62 @@ export class ConnectionStateHandler {
         this.setConnectionState(ConnectionState.Disconnected, reason);
     }
 
+    /**
+     * The "connect" event indicates the connection to the Relay Service is live.
+     * However, some additional conditions must be met before we can fully transition to
+     * "Connected" state. This function handles that interim period, known as "Connecting" state.
+     * @param connectionMode - Read or Write connection
+     * @param details - Connection details returned from the ordering service
+     */
     public receivedConnectEvent(
         connectionMode: ConnectionMode,
         details: IConnectionDetails,
     ) {
         const oldState = this._connectionState;
         this._connectionState = ConnectionState.Connecting;
+        const writeConnection = connectionMode === "write";
+
+        // Note that this may be undefined since the connection is established proactively on load
+        // and the quorum may still be under initialization.
+        const quorumClients: IQuorumClients | undefined = this.handler.quorumClients();
 
         // Stash the clientID to detect when transitioning from connecting (socket.io channel open) to connected
         // (have received the join message for the client ID)
         // This is especially important in the reconnect case. It's possible there could be outstanding
         // ops sent by this client, so we should keep the old client id until we see our own client's
-        // join message. after we see the join message for out new connection with our new client id,
+        // join message. after we see the join message for our new connection with our new client id,
         // we know there can no longer be outstanding ops that we sent with the previous client id.
         this._pendingClientId = details.clientId;
 
-        // Report telemetry after we set client id, but before transitioning to Connected state below!
+        // This pending clientId could be in the quorum already (i.e. join op already processed).
+        // We are fetching ops from storage in parallel to connecting to Relay Service,
+        // and given async processes, it's possible that we have already processed our own join message before
+        // connection was fully established.
+        // If quorumClients itself is undefined, we expect it will process the join op after it's initialized.
+        const waitingForJoinOp = quorumClients?.getMember(this._pendingClientId) === undefined;
```

Comment:
var name is a bit misleading - it should be waitingForJoinOpIfConnectingWrite.
I'd rather call it for what it checks - newClientIdNotInQuorum, or something like that

Reply:
Or && writeConnection write here (as waitingForJoinOp is used only once with that condition), then name will make sense